### PR TITLE
Improve report page in user profile

### LINKED ISSUE #22: [User settings] improve report page
The current report page is a bit broken:
image

- [x] Fix methods by using the same `clockify_seconds` methods as the frontend component
- [x] Improve a bit the design (paddings and such).
- [x] Rename the `directory` engine, the current name gives no information about whats that about. Use directories to namespace it, a possible better name could be `report`:
```
lib/decidim/time_tracker
lib/decidim/time_tracker/engine.rb
lib/decidim/time_tracker/admin_engine.rb
lib/decidim/time_tracker/report/engine.rb
```
- [x] Similary, be more agnostic in some names, for instance the controller `app/controllers/decidim/time_tracker/directory/voluntary_work_controller.rb` could be simply `app/controllers/decidim/time_tracker/report/user_controller.rb`
- [x] Add tests to this controller

### PR DESCRIPTION
**Pull request by verarojman**
Closes #22 

- [x] Spec for clockify_seconds
- [x] System spec for visiting reports page in user profile

user_reports

### CODE REVIEW

File: lib/decidim/time_tracker/reports_engine.rb [hunk 1]
```diff
@@ -5,23 +5,22 @@ module TimeTracker
...
+          root to: "user#index"
         end
       end
 
       initializer "decidim.user_menu" do
         Decidim.menu :user_menu do |menu|
-          menu.item t("voluntary_work", scope: "layouts.decidim.user_profile"),
-                    decidim_time_tracker.voluntary_work_index_path,
+          menu.item t("time_tracker", scope: "layouts.decidim.user_profile"),
+                    decidim_time_tracker.root_path,
```
Comment (microstudi) [inline lib/decidim/time_tracker/reports_engine.rb:23]: Is this correct? shouldn't this point to the users controller?

File: app/controllers/decidim/time_tracker/reports/user_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  module TimeTracker
+    module Reports
+      # The controller to handle the user's time_tracker report page.
+      class UserController < Decidim::ApplicationController
+        include Decidim::ComponentPathHelper
+        include Decidim::UserProfile
+
+        helper Decidim::TimeTracker::ApplicationHelper
+        helper_method :activities, :assignations, :total_time, :activity_path
+        # layout "layouts/decidim/application"
```
Comment (microstudi) [inline app/controllers/decidim/time_tracker/reports/user_controller.rb:13]:
```suggestion
```

File: app/models/decidim/time_tracker/assignee.rb [hunk 3]
```diff
@@ -56,6 +61,16 @@ def milestones
...
         time_events.empty?
       end
+
+      # rubocop:disable Lint/UselessAssignment
+      def self.sorted_by_status(*statuses)
+        accepted = self.accepted.sort_by(&:time_dedicated).reverse
+        pending = self.pending
+        rejected = self.rejected
+
+        statuses.map { |status| send(status) }.sum
```
Comment (microstudi) [inline app/models/decidim/time_tracker/assignee.rb:71]: code poet! cops are not nice...

**Review by microstudi — APPROVED** [review]
Just a little doubt, see the comments

File: app/models/decidim/time_tracker/assignee.rb (hunk 3 above)
Comment (verarojman) [reply app/models/decidim/time_tracker/assignee.rb:71]: All cops are bureaucrats 🙄

File: lib/decidim/time_tracker/reports_engine.rb (hunk 1 above)
Comment (verarojman) [reply lib/decidim/time_tracker/reports_engine.rb:23]: For some reason it is the only way I found to point to the correct route for the user's voluntary work (`/timetracker`)

Comment (verarojman) [reply lib/decidim/time_tracker/reports_engine.rb:23]: We can change it later if we find a more elegant solution

**Pull request merged**